### DISCUSSION TITLE
Yolo8. Train. Multi GPU

### DISCUSSION THREAD

**Discussion by Oleg-1978** [ultralytics/ultralytics#2644] [Q&A]
Hi.
I know how train Yolo5 (python -m torch.distributed.run --nproc_per_node 2 ............. device 0,1) - example.

But Yolo8 I have problem:
1. If i  train I can only use device 0,1 (model.train(epochs=300, imgsz=1024, batch=8, device='0,1'). I have no place to use _python -m torch.distributed.run --nproc_per_node 2_. And train not correctly work (time to train not reduce by two)
2. If i do go.py file (with _model.train(epochs=300, imgsz=1024, batch=8, device='0,1'_ ) and use _python -m torch.distributed.run --nproc_per_node 2 go.py_   I have same problem.

What I must do to use paralel train on Multi GPU on Yolo8?

**Comment by glenn-jocher** [comment]
@Oleg-1978 hello! Training YOLOv8 on multiple GPUs should indeed speed up the process. If you're not seeing the expected decrease in training time, there might be an issue with how the distributed training is set up.

For YOLOv8, you should be able to specify multiple GPUs directly in the training command without needing to use `torch.distributed.run`. The framework is designed to automatically detect and use all available GPUs when you set `device='0,1'` (for using GPU 0 and GPU 1, for example). If you have more than two GPUs, you can expand this to match the number of GPUs you want to use, like `device='0,1,2,3'`.

Make sure that:
- You have the latest version of YOLOv8 from the repo.
- Your environment is set up correctly with all dependencies installed.
- You're using the correct syntax for the training command in YOLOv8.

If you continue to face issues, please ensure that your system's CUDA and cuDNN are properly configured, and that PyTorch can access all GPUs. You can check accessible GPUs using `torch.cuda.device_count()` in a Python shell.

If the problem persists, consider opening an issue on the repo with detailed information about your setup and the exact behavior you're observing. The community and the Ultralytics team can then help troubleshoot the issue more effectively. Remember to check the documentation at https://docs.ultralytics.com for any updates or detailed instructions on multi-GPU training. 😊
